repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/InspectProcessDuration.java

```diff
@@ -17,18 +17,19 @@
     private EndProcessNotifier endProcessNotifier;
     private InspectBuilds inspectBuilds;
     private InspectJobs inspectJobs;
-    private DockerPipelineRunner pipelineRunner;
+    private DockerPipelineRunner DockerPipelineRunner;
```

Comment:
prefer having one single field `runner` typed with a common abstraction (super type) on top of both DockerPipelineRunner and ActiveMQPipelineRunner

Reply:
Per discussion yesterday, I mentioned that I tried to do this according to your proposal, but it make things complicated to change the DockerPipelineRunner structure. The problem is because RTScanner overloads the  `submitBuild` method in `DockerPoolManager` which is already implemented the interface `PipelineRunner` if follows your suggested structure, so we would violate the "already defined in DockerPool" interface contract if we do that. How about we change the name of the interface PipelineRunner to NonDockerPipelineRunner for future non-docker implementation and if a class want to run with docker they can extends from Dockerpool as usual ? :) .